kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/ContinuousPlanning.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.server.services.optaplanner;
+
+import java.util.Collection;
+import java.util.List;
+
+import org.optaplanner.core.api.solver.event.SolverEventListener;
+import org.optaplanner.core.impl.solver.ProblemFactChange;
+
+/**
+ * Provides integration for OptaPlanner to continuously solve planning problems 
+ * based on support of adding problem facts.
+ *
+ */
+public interface ContinuousPlanning {
```

Comment:
I think we should consider using term 'Real-time planning' here. We're aiming for a solving process, which will run without interruptions (as opposed to a regular task). I have to admit the terms are a bit confusing and the difference may not be obvious at first sight. I'll let @ge0ffrey add more insight.